exercises/hangman/src/test/java/HangmanTest.java

```diff
@@ -0,0 +1,200 @@
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.disposables.Disposable;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Stream;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class HangmanTest {
+
+    private Hangman hangman;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void init() {
+        hangman = new Hangman();
+    }
+
+    @Test
+    public void initialization() {
+        Observable<Output> result = hangman.play(
+            Observable.fromArray("secret"),
+            Observable.fromArray());
+        Output init = result.blockingFirst();
+        assertNotNull(init);
+        assertEquals("secret", init.secret);
+        assertEquals("______", init.discovered);
+        assertEquals(Collections.emptySet(), init.guess);
+        assertEquals(Collections.emptySet(), init.misses);
+        assertEquals(Collections.emptyList(), init.parts);
+        assertEquals(Status.PLAYING, init.status);
```

Comment:
If you override equals and hashCode, you can just assertEquals the whole object. Just a thought, although this is acceptable too.

Reply:
Yes for an actual application. But in this case, I wanted to explicit the expected output.
It is more tedious for the test writer to explicit all cases but it is better for the user playing with the exercise to have an explicit list. This way, you don't have to play with the `toString` of the objects to find where is your error.